src/Storages/Kafka/KafkaConsumer.h

```diff
@@ -113,11 +117,20 @@ class KafkaConsumer
         rdkafka_stat = stat_json_string;
     }
     void inUse() { in_use = true; }
-    void notInUse() { in_use = false; }
+    void notInUse()
+    {
+        in_use = false;
+        last_used_usec = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
```

Comment:
not very obvious choice of moment of setting last_used time. 

Why not inside KafkaConsumer::consume? (it's even have last poll time already)

Reply:
last_poll_time is not the same, it can differs from the last_used_usec, because, after poll had been finished other things could be done:
- reading from this source could take some time
- commit
- writing this data to MV could take some time

And updating it in `notInUse` looks like the most logical, since KafkaConsumer should know nothing about TTL (but to avoid tracking this timestamps in a separate map, this time had been added to the KafkaConsumer)

Reply:
OK. Maybe the option to convert `consumers` from StorageKafka into class with methods pop / push, also maintaining that `busy` / `in_use` flag & time of the last usage there would be less confusing (since KafkaConsumer class don't do anything with that anyway). But OK.